src/pan3d/widgets/color_by.py

```diff
@@ -0,0 +1,276 @@
+from vtkmodules.vtkCommonCore import vtkLookupTable
+
+from pan3d.utils.convert import to_image
+from pan3d.utils.presets import PRESETS, set_preset
+from trame.widgets import html
+from trame.widgets import vuetify3 as v3
+
+
+class ColorBy(html.Div):
+    """
+    Color settings for the XArray Explorers.
+    """
+
+    _next_id = 0
+
+    @classmethod
+    def next_id(cls):
+        """Get the next unique ID for the scalar bar."""
+        cls._next_id += 1
+        return f"pan3d_scalarbar{cls._next_id}"
+
+    def __init__(
+        self,
+        data_arrays=None,
```

Comment:
what is the shape of data_arrays?

Reply:
I'd like a blank list `data_arrays=[]`
However, with the new pre-commit changes, we are dinged. 
```
src/pan3d/widgets/color_by.py:24:21: B006 Do not use mutable data structures for argument defaults
   |
22 |     def __init__(
23 |         self,
24 |         data_arrays=[],
   |                     ^^ B006
25 |         color_by=None,
26 |         preset="Fast",
   |
   = help: Replace with `None`; initialize within function

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).

```

Reply:
yes that is because you should not add a mutable object as default. Doing that will introduce bug.

The way to do it properly in Python is as follow

```
def abs(var=None):
      if var is None:
          var = []
```

Reply:
But still not telling me the content of data_arrays

data_arrays = [
    { name: "hello", range: [10, 20]}, # ??
    { title: "hello", min: 10, max: 20}, # ??
    else ???
   
]

Reply:
still relevant and added similar comment today as well.